src/openrct2-ui/windows/NewsOptions.cpp

```diff
@@ -82,212 +84,203 @@ static rct_widget window_news_options_widgets[] = {
-    1,
-    4,
-    4,
-};
-const int32_t window_news_option_tab_animation_frames[] = {
-    1,
-    16,
-    8,
-};
+private:
+    void SetPage(int32_t p)
+    {
+        if (this->page != p)
+        {
+            this->page = p;
+            this->frame_no = 0;
+            this->Invalidate();
+        }
+    }
 
-static void WindowNewsOptionsDrawTabImage(rct_window* w, rct_drawpixelinfo* dpi, int32_t page, int32_t spriteIndex)
-{
-    rct_widgetindex widgetIndex = WIDX_TAB_PARK + page;
+    void DrawTabImages(rct_drawpixelinfo* dpi)
+    {
+        DrawTabImage(dpi, NOTIFICATION_CATEGORY_PARK, SPR_TAB_PARK);
+        DrawTabImage(dpi, NOTIFICATION_CATEGORY_RIDE, SPR_TAB_RIDE_0);
+        DrawTabImage(dpi, NOTIFICATION_CATEGORY_GUEST, SPR_TAB_GUESTS_0);
+    }
 
-    if (!(w->disabled_widgets & (1LL << widgetIndex)))
+    void DrawTabImage(rct_drawpixelinfo* dpi, int32_t p, int32_t spriteIndex)
     {
-        if (w->page == page)
+        rct_widgetindex widgetIndex = WIDX_TAB_1 + p;
+
+        if (!(disabled_widgets & (1LL << widgetIndex)))
         {
-            int32_t numFrames = window_news_option_tab_animation_frames[w->page];
-            if (numFrames > 1)
+            if (this->page == p)
             {
-                int32_t frame = w->frame_no / window_news_option_tab_animation_divisor[w->page];
-                spriteIndex += (frame % numFrames);
+                int32_t numFrames = window_news_option_tab_animation_frames[this->page];
+                if (numFrames > 1)
+                {
+                    int32_t frame = this->frame_no / window_news_option_tab_animation_divisor[this->page];
+                    spriteIndex += (frame % numFrames);
+                }
             }
+
+            const auto& widget = this->widgets[widgetIndex];
+            gfx_draw_sprite(dpi, ImageId(spriteIndex), this->windowPos + ScreenCoordsXY{ widget.left, widget.top });
         }
+    }
 
-        const auto& widget = w->widgets[widgetIndex];
-        gfx_draw_sprite(dpi, ImageId(spriteIndex), w->windowPos + ScreenCoordsXY{ widget.left, widget.top });
+    void SetPressedTab()
+    {
+        for (int32_t i = 0; i < NOTIFICATION_CATEGORY_COUNT; i++)
+            this->pressed_widgets &= ~(1 << (WIDX_TAB_1 + i));
+        this->pressed_widgets |= 1LL << (WIDX_TAB_1 + this->page);
     }
-}
 
-static void WindowNewsOptionsDrawTabImages(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    WindowNewsOptionsDrawTabImage(w, dpi, NOTIFICATION_CATEGORY_PARK, SPR_TAB_PARK);
-    WindowNewsOptionsDrawTabImage(w, dpi, NOTIFICATION_CATEGORY_RIDE, SPR_TAB_RIDE_0);
-    WindowNewsOptionsDrawTabImage(w, dpi, NOTIFICATION_CATEGORY_GUEST, SPR_TAB_GUESTS_0);
-}
+    bool* GetNotificationValuePtr(const NotificationDef* ndef)
+    {
+        bool* configValue = reinterpret_cast<bool*>(reinterpret_cast<size_t>(&gConfigNotifications) + ndef->config_offset);
+        return configValue;
+    }
+
+    const int32_t window_news_option_tab_animation_divisor[3] = {
```

Comment:
as per the coding style `Local constants do not require underscore prefix. They are title case` could you make these two title case.